fragalysis_api/xcimporter/align.py

```diff
@@ -98,138 +111,407 @@ def __get_length_and_resolution(self, file):
-                wanted_ligs.append(lig[16:20] + lig[20:26])
-                # print(lig[16:20].strip() + lig[20:26])
-
-        wanted_ligs = list(set(wanted_ligs))
-
-        return wanted_ligs
-
-    def save_chain(self, lig, f):
-        lig_chain = lig[5]
-        name = os.path.splitext(os.path.basename(f))[0] + '_' + str(lig_chain)
-        filename = os.path.join(self.outdir, f'{name}_mono.pdb')
-        pymol.cmd.load(f, name)
-        pymol.cmd.save(filename, f'chain {lig_chain}')
-        pymol.cmd.reinitialize()
-
-        return filename
-
-    def process_ligs(self, filename):
-        test_block = open(filename, 'r').readlines()
-        ligs = self.find_ligs(test_block)
-        print(ligs)
-        outnames = []
-        for lig in ligs:
-            o = self.save_chain(lig, filename)
-            outnames.append(o)
+    def get_maplist(self):
+        '''
+        Get a list of .map or .ccp4 files from within the input directory
+        :return: Returns a list of filepaths corresponding to said volume density files.
+        '''
+        map_files = glob.glob(os.path.join(self.directory, '*.map'))
+        cpp4_files = glob.glob(os.path.join(self.directory, '*.ccp4'))
+        return map_files + cpp4_files
+
+    def split_chains(self, f):
+        '''
+        Split a pdb file according to chain names. While preserving chains that contain waters and assigned small, non-amino acid containing chains to the nearest chain by center of mass.
+        :param f: File name of the pdb file to split
+        :return: A list of pdb_files that have been created with the naming convention f_[chain_name]
+        '''
+        aa_codes = {'V': 'VAL', 'I': 'ILE', 'L': 'LEU', 'E': 'GLU', 'Q': 'GLN', 'D': 'ASP', 'N': 'ASN', 'H': 'HIS',
+                        'W': 'TRP', 'F': 'PHE', 'Y': 'TYR', 'R': 'ARG', 'K': 'LYS', 'S': 'SER', 'T': 'THR', 'M': 'MET',
+                        'A': 'ALA', 'G': 'GLY', 'P': 'PRO', 'C': 'CYS'}
+        filenames = []
+        base_structure = gemmi.read_structure(f)
+        base_models = base_structure[0]
+        all_chains = [x.name for x in base_models]
+        HOH_chains = find_water_chains(f)
+
+        nonHOH_chains = list(set(all_chains) - set(HOH_chains))
+
+        chain_centers = {}
+        chain_names = []
+        # From nonHOH chains, identify if they contain any aminoacids, if they do. Assume they are full chains.
+        # else, later on they will be attached to nearest chain by center.
+        for i in nonHOH_chains:
+            chain_centers[i] = get_chain_center(chain_name=i, file=f)
+            span = base_structure[0][i].whole()
+            residues = [x.name for x in span]
+            # Possible to convert this to a %age
+            if any([True for v in residues if v in aa_codes.values()]):
+                chain_names.append(i)
+
+        alt_chains = list(set(nonHOH_chains) - set(chain_names))
+        for i in chain_names:
```

Comment:
can probably un-nest this a bit?Combine I and j into a list of tuples or something? - not important now